core/commands/repository/interactors/encode_secret_string.py

```diff
@@ -0,0 +1,27 @@
+from dataclasses import dataclass
+
+from codecov.commands.base import BaseInteractor
+from codecov.commands.exceptions import Unauthenticated, ValidationError
+from codecov.db import sync_to_async
+from codecov_auth.models import Owner
+from core.commands.repository.interactors.utils import encode_secret_string
+from core.models import Repository
+
+
+class EncodeSecretStringInteractor(BaseInteractor):
+    @sync_to_async
+    def execute(self, owner: Owner, repo: Repository, value: str) -> str:
+        if not self.current_user.is_authenticated:
+            raise Unauthenticated()
+        if not repo:
+            raise ValidationError("Repo not found")
```

Comment:
1. What does the encoded secret string actually do?
2. Currently this allows authenticated users not belonging to the org to generate this value as well, is that ok?

Reply:
We have a widget in gazebo that lets us generate a secret string(for any text, but meant to be used for YAML values) 

<img width="748" alt="Screenshot 2024-05-19 at 2 35 31 PM" src="https://github.com/codecov/codecov-api/assets/148245014/bd63b923-4810-45fc-9380-7109dbb9072f">

It returns an encrypted string for the text entered. It's basically a big wrapper around the `encode_secret_string` function. 

Since the user can enter any string, and they will need org validation to see other sensitive data, I am comfortable with allowing any authenticated user to call this. 

Let me know if you think I'm missing anything, thanks.

Reply:
Gotcha, as long as this has the same auth validation as the internal REST endpoint its fine.